### ISSUE TITLE
Hab-sup as pid1 in docker container

### ISSUE THREAD

**Issue by dmajere** [habitat-sh/habitat#1606]
Hello,

```
hab --version
hab 0.15.0/20161222203215

hab-sup --version
hab-sup 0.15.0/20161222205412
```

hab-sup is running as pid1 in docker container and it should do child reaping, because otherwise we are ending with lots of zombies inside container

**Comment by fnichol** [comment]
@dmajere Hi, the Supervisor *should* reap zombie process, however that might not be happening in practice. Would you be able to provide an example of this so we can verify? Thanks!

**Comment by adamhjk** [comment]
@fnichol I suspect this got undone in the windows refactor - in particular, all the wait logic appears to have been moved into core, and then abstracted. The end result is I don't see us doing the right thing inside the manager right now. We should put this up high on the triage priority.

**Comment by adamhjk** [comment]
It may have not worked properly from longer back than that.

**Comment by mwrock** [comment]
I remember looking at the zombie reaping code and in its state at the time just before the windows refactor, it could not have been working. I left fairly detailed notes in the [PR](https://github.com/habitat-sh/habitat/pull/1577).

**Comment by smurawski** [comment]
There is still a bit of churn in the supervisor code base as we move to a multi-tenant supervisor.  We'll have to make sure this behavior lands as part of that effort.

/cc @fnichol @habitat-sh/habitat-core-maintainers

**Comment by reset** [comment]
I haven't been able to reproduce this behaviour in the latest supervisor. The multi-tenant supervisor has landed in 0.20.0 and I'd give that a shot and re-open if you are still having issues!
